crates/types/Cargo.toml

```diff
@@ -20,6 +20,7 @@ serialize_hierarchy = { workspace = true }
 spl_network_messages = { workspace = true }
 tokio = { workspace = true }
 bincode = { workspace = true }
+filtering = { workspace = true }
```

Comment:
I was also not sure about this. On the one hand, it removes tons of code duplication. I am not a fan of the filtering snap shot, as it is a exact duplication of the filters internal state.
I also thought about the following approach, feel free to give your opinion on that:

Extract all state structs to the types crate and implement extension traits to support kalman updates on these state structs. This way, the kalman filter would define a common interface and can be implemented for different underlying datastructures (which is currently not needed, and thus yagni).
The effect would be the same, with the difference of the types crate not depending or including any implementation code, as it is currently the case.

Do you have other ideas? We should somehow be able to somehow expose third-party types (and this also means creates in `hulk/crates/`) to communication. But I am not sure how to solve this. The IMHO best solution would be to move the code generation to a place, where the same dependencies are available to the node implementation and the code generation (as done in #143 ). This way we would be able to generate for any type, that fulfills the necessary trait bounds.

Reply:
To conclude, I propose to implement the first suggestion in this iteration: 
splitting implementation and structs to different crates

The other stuff may be future work in a future iteration to the framework like #143

Reply:
Ok, fine for me. We still have the dependency that `filtering` depends on `types`.

Reply:
I'm not sure how to resolve that. I think it is fine for now. As I said, it would be nice to use any type from any crate as an input/output in our framework -> future work

So this discussion is resolved?

Reply:
Understood, yes, resolved because it is already resolved.